gadgets/trace_lsm/program.bpf.c

```diff
@@ -0,0 +1,228 @@
+	F(file_alloc_security)      \
+	F(file_free_security)       \
+	F(file_ioctl)               \
+	F(file_ioctl_compat)        \
+	F(mmap_addr)                \
+	F(mmap_file)                \
+	F(file_mprotect)            \
+	F(file_lock)                \
+	F(file_fcntl)               \
+	F(file_set_fowner)          \
+	F(file_send_sigiotask)      \
+	F(file_receive)             \
+	F(file_open)                \
+	F(task_alloc)               \
+	F(task_free)                \
+	F(cred_alloc_blank)         \
+	F(cred_free)                \
+	F(cred_prepare)             \
+	F(cred_transfer)            \
+	F(cred_getsecid)            \
+	F(kernel_act_as)            \
+	F(kernel_create_files_as)   \
+	F(kernel_module_request)    \
+	F(kernel_load_data)         \
+	F(kernel_post_load_data)    \
+	F(kernel_read_file)         \
+	F(kernel_post_read_file)    \
+	F(task_fix_setuid)          \
+	F(task_fix_setgid)          \
+	F(task_setpgid)             \
+	F(task_getpgid)             \
+	F(task_getsid)              \
+	F(task_getsecid_obj)        \
+	F(task_setnice)             \
+	F(task_setioprio)           \
+	F(task_getioprio)           \
+	F(task_prlimit)             \
+	F(task_setrlimit)           \
+	F(task_setscheduler)        \
+	F(task_getscheduler)        \
+	F(task_movememory)          \
+	F(task_kill)                \
+	F(task_prctl)               \
+	F(task_to_inode)            \
+	F(ipc_permission)           \
+	F(ipc_getsecid)             \
+	F(msg_msg_alloc_security)   \
+	F(msg_msg_free_security)    \
+	F(msg_queue_alloc_security) \
+	F(msg_queue_free_security)  \
+	F(msg_queue_associate)      \
+	F(msg_queue_msgctl)         \
+	F(msg_queue_msgsnd)         \
+	F(msg_queue_msgrcv)         \
+	F(shm_alloc_security)       \
+	F(shm_free_security)        \
+	F(shm_associate)            \
+	F(shm_shmctl)               \
+	F(shm_shmat)                \
+	F(sem_alloc_security)       \
+	F(sem_free_security)        \
+	F(sem_associate)            \
+	F(sem_semctl)               \
+	F(sem_semop)                \
+	F(netlink_send)             \
+	F(d_instantiate)            \
+	F(getprocattr)              \
+	F(setprocattr)              \
+	F(ismaclabel)               \
+	F(secid_to_secctx)          \
+	F(secctx_to_secid)          \
+	F(release_secctx)           \
+	F(inode_invalidate_secctx)  \
+	F(inode_notifysecctx)       \
+	F(inode_setsecctx)          \
+	F(inode_getsecctx)
+
+#define ENUM_ITEM(name) name,
+
+enum lsm_tracepoint { FOR_EACH_LSM_HOOK(ENUM_ITEM) };
+
+struct event {
+	gadget_mntns_id mntns_id;
+	gadget_timestamp timestamp;
+	u32 pid;
+	u32 tid;
+	u32 uid;
+	u32 gid;
+	enum lsm_tracepoint tracepoint;
+	char comm[TASK_COMM_LEN];
+};
+
+GADGET_TRACER_MAP(events, 1024 * 256);
+
+GADGET_TRACER(lsm, events, event);
+
+#define DECLARE_LSM_PARAMETER(name)               \
+	const volatile bool trace_##name = false; \
+	GADGET_PARAM(trace_##name);
+
+const volatile bool trace_all = false;
+GADGET_PARAM(trace_all);
```

Comment:
I wonder if this should be true by default. Otherwise, when running the gadget without CLI flags, it will do nothing.

Maybe that will just be something to explain in the README.md later on.